src/components/Hoverable/index.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * It is necessary to create a Hoverable component instead of relying solely on Pressable support for hover state,
+ * because nesting Pressables causes issues. https://github.com/necolas/react-native-web/issues/1875
+ */
+import React, {Component} from 'react';
+import {propTypes, defaultProps} from './HoverablePropTypes';
+
+class Hoverable extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            isHovered: false,
+        };
+        this.handleMouseEnter = this.handleMouseEnter.bind(this);
+        this.handleMouseLeave = this.handleMouseLeave.bind(this);
+    }
+
+    handleMouseEnter() {
+        if (!this.state.isHovered) {
+            this.props.onHoverIn();
+            this.setState({isHovered: true});
+        }
+    }
+
+    handleMouseLeave() {
+        if (this.state.isHovered) {
+            this.props.onHoverOut();
+            this.setState({isHovered: false});
+        }
+    }
+
+    render() {
+        const child = typeof this.props.children === 'function'
+            ? this.props.children(this.state.isHovered)
+            : this.props.children;
+        return React.cloneElement(React.Children.only(child), {
+            onMouseEnter: this.handleMouseEnter,
+            onMouseLeave: this.handleMouseLeave,
+        });
```

Comment:
It took me a second to follow what is happening here so going to rubber duck

1. This component renders a function as a child
2. The function passes an `isHovered` state
3. `onMouseEnter` and `onMouseLeave` are passed to the child
4. That updates the state of this component and re-renders the child passing `isHovered` to it

This is interesting. 

Just for reference I maybe would have done it this way... put your `onHoverIn` and `onHoverOut` directly on the `Hoverable` when used and maybe just default it to a `View` on web instead of doing the `React.Children` stuff. 

That way the hover state doesn't need to be encapsulated in the `Hoverable` and you can do whatever you want to do `onHoverIn` or `onHoverOut`. Not saying the way you did this is bad (might even be preferable since it doesn't trigger a re-render on the parent component), but wanted to throw out this alternative

```jsx
<Hoverable
    onHoverIn={set to parent state}
    onHoverOut={set to parent state}
>
    <Anything showThing={this.state.showThing} />
</Hoverable>
```

Not as fancy, but might be easier to understand.

Reply:
Ok, returning to this one since I'm seeing that maybe you want to go for a similar vibe to what `Pressable` is doing and it's making a bit more sense now..

Reply:
Yeah, it is pretty similar to the interface for `Pressable`. FWIW, this `Hoverable` implementation was prettymuch a simplified version of [this gist](https://gist.github.com/necolas/1c494e44e23eb7f8c5864a2fac66299a), which was created by the author of `react-native-web`.